backpack/extensions/firstorder/batch_l2_grad/convtransposend.py

```diff
@@ -1,23 +1,64 @@
+"""batch_l2 extension for ConvTranspose."""
 from torch import einsum
 
-from backpack.extensions.firstorder.base import FirstOrderModuleExtension
+from backpack.core.derivatives.conv_transposend import ConvTransposeNDDerivatives
+from backpack.extensions.firstorder.batch_l2_grad.batch_l2_base import BatchL2Base
 from backpack.utils import conv_transpose as convTransposeUtils
 
 
-class BatchL2ConvTransposeND(FirstOrderModuleExtension):
+class BatchL2ConvTransposeND(BatchL2Base):
+    """batch_l2 extension for ConvTranspose."""
+
     def __init__(self, N, params=None):
-        super().__init__(params=params)
+        """Initialization.
+
+        Args:
+            N: number of dimensions
+            params: list of parameters. Defaults to None.
+        """
         self.N = N
+        super().__init__(
+            params=params, derivatives=ConvTransposeNDDerivatives(N=self.N)
```

Comment:
`N` seems to only be used to determine which derivative should be used -> Replace (`ConvTransposeNDDerivatives` + `N`) by explicity `ConvTranspose{1,2,3}d`?

Reply:
**Update:** The above comment is incorrect. as `self.N` is used further below as argument of `convTransposeUtils.getweight_gradient_factors`. Could you take a look how difficult it would be to eliminate `N` from the arguments of `convTransposeUtils.getweight_gradient_factors`? I recently did a very similar refactoring on those files, however for the second-order, rather than the first-order extensions.

Reply:
I removed `N` and refactored.